File: 02/py/02.py
#!/usr/bin/env python3

import re

data = [(lambda a,b,c,d: [int(a), int(b), c, d])(*re.findall(r'\w+', line))
        for line in open('input').readlines()]

# part 1
print(sum(a <= d.count(c) <= b for a,b,c,d in data))

# part 2
print(sum((d[a-1] == c) != (d[b-1] == c) for a,b,c,d in data))

File: 02/rb/02.rb
#!/usr/bin/env ruby

input = File.readlines('input').map{|line| a,b,c,d = line.scan /\w+/; [a.to_i, b.to_i, c, d]}

# part 1
p input.count{|a,b,c,d| (a..b).include? d.count c}

# part 2
p input.count{|a,b,c,d| (d[a-1] == c) != (d[b-1] == c)}
